packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -86,4 +134,131 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
+                        },
+                    },
+                };
+                await this._store.dispatch(this._paymentActionCreator.submitPayment({ methodId, paymentData }));
+                window.location.assign('/checkout/order-confirmation');
+            }
+        } catch (e) {
+            throw new RequestError(e);
+        }
+    }
+
+    private async _onShippingChangeHandler(data: ShippingOptionsPayload) {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const { id: selectedShippingOptionId } = data.selected_shipping_option || {};
+        const shippingAddress = await this._transformToAddress(data.shipping_address);
+        this._currentShippingAddress = shippingAddress;
+        const lineItems = this._getLineItems();
+        const consignment = { shippingAddress, lineItems };
+        const existingConsignments = state.consignments.getConsignmentsOrThrow();
+        if (existingConsignments[0]) {
+            await this._store.dispatch(this._consignmentActionCreator.updateConsignment({id: existingConsignments[0].id, ...consignment}));
+        } else {
+            await this._store.dispatch(this._consignmentActionCreator.createConsignments([consignment]));
+        }
+        const updatedConsignment = state.consignments.getConsignmentsOrThrow()[0];
+        const { availableShippingOptions } = updatedConsignment;
+        const { id: recommendedShippingOptionId } = availableShippingOptions?.find(option => option.isRecommended) || {};
+        const isSelectedOptionExist = selectedShippingOptionId && availableShippingOptions?.find(option => option.id === selectedShippingOptionId);
+        await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+        const shippingOptionId = isSelectedOptionExist ? selectedShippingOptionId : recommendedShippingOptionId;
+
+        if (shippingOptionId) {
+            await this._store.dispatch(this._consignmentActionCreator.selectShippingOption(
+                shippingOptionId
+            ));
+        }
+
+        if (availableShippingOptions) {
+            await this._paypalCommercePaymentProcessor.setShippingOptions({
+                ...data,
+                cartId: cart.id,
+                availableShippingOptions,
+            });
+        }
+    }
+
+    private async _transformToAddress({city, postal_code, country_code}: ShippingAddress) {
+        const state = this._store.getState();
+        const existingConsignment = state.consignments.getConsignmentsOrThrow();
+        const billingAddress = state.billingAddress.getBillingAddressOrThrow();
+        const { email } = billingAddress;
+
+        const { firstName, lastName, address1 } = existingConsignment[0]?.shippingAddress || {};
+        const shippingData = {
+            city: city,
+            postalCode: postal_code,
+            countryCode: country_code,
+            firstName: firstName || 'Fake',
```

Comment:
Does it make sense to have fallback values as `Fake`? Can we not just have `''` ?

Reply:
It is should not be empty

Reply:
So are we creating a `Fake` consignment as part of `onShippingChange`?

Reply:
Yes